### PR TITLE
Prevent data duplication issues

### PR DESCRIPTION
**Pull request by dlsf**
- [x] Add unique restraints to all tables
- [x] Add a way to update the database (patch system)
- [x] Test the patch system
- [x] Add unique constraint patch for multiple columns per table
- [x] Fix connection leaks
- [x] Update Gradle because it's better
- [x] Refactor random stuff
- [x] Testing. An exception might be thrown if OrmLite tries to insert data that violates the unique restraint

### CODE REVIEW

**Review by PeachesMLG — APPROVED** [review]

File: src/main/java/com/iridium/iridiumskyblock/database/IslandBan.java [hunk 1]
```diff
@@ -17,13 +17,13 @@
 @DatabaseTable(tableName = "island_bans")
 public class IslandBan extends IslandData {
 
-    @DatabaseField(columnName = "id", generatedId = true, canBeNull = false)
+    @DatabaseField(columnName = "id", generatedId = true, canBeNull = false, unique = true, uniqueCombo = true)
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/database/IslandBan.java:20]: i've tried adding columns but its throws errors, so Orm doesn't recreate|update tables if is exist, ik this not a new column but i guess orm wont update the table 
But u're god -_-

**Review by MertUnverdi — APPROVED** [review]
Das is better so always he right 😅

File: src/main/java/com/iridium/iridiumskyblock/DataConverter.java [hunk 2]
```diff
@@ -1,66 +1,87 @@
...
+        IridiumSkyblock.getInstance().getDatabaseManager().getIslandBankTableManager().deleteDuplicates();
+        IridiumSkyblock.getInstance().getDatabaseManager().getIslandBlocksTableManager().deleteDuplicates();
+        IridiumSkyblock.getInstance().getDatabaseManager().getIslandBoosterTableManager().deleteDuplicates();
+        IridiumSkyblock.getInstance().getDatabaseManager().getIslandMissionTableManager().deleteDuplicates();
+        IridiumSkyblock.getInstance().getDatabaseManager().getIslandSpawnersTableManager().deleteDuplicates();
+        IridiumSkyblock.getInstance().getDatabaseManager().getIslandUpgradeTableManager().deleteDuplicates();
+    }
+
+    public static void updateDatabaseData(int oldVersion, int newVersion, ConnectionSource connectionSource) {
+        System.out.println("Updating database from version " + oldVersion + " to " + newVersion);
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/DataConverter.java:66]: Any reason why we're not using getLogger?

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/DataConverter.java:66]: its same -_-

**Comment by PeachesMLG** [comment]
Fixes: https://github.com/Iridium-Development/IridiumSkyblock/issues/373

**Comment by dlsf** [comment]
> Fixes: #373

Does it?

**Comment by PeachesMLG** [comment]
> > Fixes: #373
> 
> Does it?

Ye cuz u added Armor stand to the entity list stoopid

**Comment by dlsf** [comment]
That should prevent Armorstands from being placed but not from being interacted with

**Comment by PeachesMLG** [comment]
> That should prevent Armorstands from being placed but not from being interacted with

oh, well whatever the issue should be fixed anyways

**Comment by dlsf** [comment]
> oh, well whatever the issue should be fixed anyways

I tested it yesterday and it wasn't fixed

**Comment by PeachesMLG** [comment]
> > oh, well whatever the issue should be fixed anyways
> 
> I tested it yesterday and it wasn't fixed

Stop it with your completely logical arguments >:C

**Pull request merged**